tools/docs/generate2.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""A tool to generate api_docs for TensorFlow Addons.
+
+```
+Run from root folder of source code
+python tools/docs/generate2.py --output_dir=docs/
+```
+
+Requires a local installation of:
+  Docs Generator - pip install git+https://github.com/tensorflow/docs
+  tensorflow_addons
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from os import path
+
+from absl import app
+from absl import flags
+import tensorflow_addons as tfa
+
+from tensorflow_docs.api_generator import doc_controls
+from tensorflow_docs.api_generator import generate_lib
+from tensorflow_docs.api_generator import parser
+
+from tensorflow.python.util import tf_inspect
+
+# Use tensorflow's `tf_inspect`, which is aware of `tf_decorator`.
+parser.tf_inspect = tf_inspect
+
+# So patch `tfa.__all__` to list everything.
+tfa.__all__ = [item_name for item_name, value in tf_inspect.getmembers(tfa)]
```

Comment:
This line is likely not necessary. 

The doc generator respects a module's `__all__` but will document everything if __all__ is not set. I only added this for tensorflow/tensorflow because there was a miss-match between what was in __all__ and what we wanted to document.

Reply:
thanks.. will remove it.